scheduling/task_run.go

```diff
@@ -0,0 +1,160 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskRunId int32
+type TaskId int32 // TODO: This datatype already exists in rasks.go, so when we merge, we should use the same?
+type Status string
+
+const (
+	Success Status = "SUCCESS"
+	Failed  Status = "FAILED"
+	Pending Status = "PENDING"
+)
+
+type TriggerType string
+
+const (
+	oneOffTrigger TriggerType = "OneOffTrigger"
+	dummyTrigger  TriggerType = "DummyTrigger"
+)
+
+type Trigger interface {
+	Type() TriggerType
+	Name() string
+}
+
+type OneOffTrigger struct {
+	TriggerName string      `json:"triggerName"`
+	TriggerType TriggerType `json:"triggerType"`
+}
+
+func (t OneOffTrigger) Type() TriggerType {
+	return t.TriggerType
+}
+
+func (t OneOffTrigger) Name() string {
+	return t.TriggerName
+}
+
+type DummyTrigger struct {
+	TriggerName string      `json:"triggerName"`
+	TriggerType TriggerType `json:"triggerType"`
+	DummyField  bool        `json:"dummyField"`
+}
+
+func (t DummyTrigger) Type() TriggerType {
+	return t.TriggerType
+}
+
+func (t DummyTrigger) Name() string {
+	return t.TriggerName
+}
+
+type TaskRunMetadata struct {
+	ID        TaskRunId `json:"id"`
+	RunId     TaskId    `json:"runId"`
+	Name      string    `json:"name"`
+	Trigger   Trigger   `json:"trigger"`
+	Status    Status    `json:"status"`
+	StartTime time.Time `json:"startTime"`
+	EndTime   time.Time `json:"endTime"`
+	Logs      []string  `json:"logs"`
+	Error     string    `json:"error"`
+}
+
+// Read Methods
+// func (t *TaskRunMetadata) ID() TaskRunId
+// func (t *TaskRunMetadata) RunID() TaskId
+// func (t *TaskRunMetadata) Name() string
+// func (t *TaskRunMetadata) Trigger() Trigger
+// func (t *TaskRunMetadata) Status() Status // Enum
+// func (t *TaskRunMetadata) StartTime() Time
+// func (t *TaskRunMetadata) EndTime() Time
+// func (t *TaskRunMetadata) Logs() []string
+// func (t *TaskRunMetadata) Error() string
+
+// Formatting
+func (t *TaskRunMetadata) ToJSON() ([]byte, error) {
```

Comment:
Are we not going to use the traditional marshalling method names we talked about?